docs/rfc/rfc-7.adoc

```diff
@@ -0,0 +1,288 @@
+must be confirmed as already mined.
+
+==== Benchmarks
+
+The most computationally expensive part of the signing protocol takes 2.08s
+on a 10-core Apple M1 machine for a single signing group member. Assuming the
+client works on a 2-core machine and there are no more than 5 signing group
+members on a single machine, the bottleneck phase should take no more than
+1 minute.
+
+The announcement phase takes always 7 blocks, so ~1.5 minutes assuming 12s block
+time. With 30 blocks timeout for a single attempt, it leaves more than 3 minutes
+for the rest of the signing protocol which should be more than enough.
+This should also be enough for local development when all signing group members
+reside on the same computer. Based on the local benchmarks, the entire signing
+never takes more than 5 minutes locally, and usually, it takes less than 4
+minutes.
+
+Assuming it takes no more than 4 minutes in the happy path to sign a single
+input, it should take no more than an hour to sign 15 inputs. Given that the
+entire sweep operation - including 6 Bitcoin confirmations - should complete
+within 4 hours, it gives just enough time for a single Bitcoin miner fee bump.
+
+==== Retries
+
+
+The signing protocol is non-attributable and we need to retry in case of
+corrupted data:
+
+- With 1 malicious member in a signing group, we need 2 attempts of the protocol
+in the worst case (`P = (99 choose 51) / (100 choose 51) = 0.49`).
+In the worst case, it takes approximately 2 hours to sign 15 inputs.
+
+- With 2 malicious members in a signing group, we need 5 attempts of the protocol
+in the worst case (`P = (98 choose 51) / (100 choose 51) = 0.2375757575`).
+In the worst case, it takes approximately 7h 30min to sign 15 inputs.
+
+- With 3 malicious members in a signing group, we need 10 attempts of the protocol
+in the worst case (`P = (97 choose 51) / (100 choose 51) = 0.1139393939`).
+In the worst case, it takes approximately 15 hours to sign 15 inputs.
+
+- With 4 malicious members in a signing group, we need 20 attempts of the
+protocol in the worst case (`P = (96 choose 51) / (100 choose 51) = 0.0540331146`).
+In the worst case, it takes approximately 30 hours to sign 15 inputs.
+
+- With 5 malicious members in a signing group, we need 40 attempts of the protocol
+in the worst case (`P = (95 choose 51) / (100 choose 51) = 0.0253280224`).
+In the worst case, it takes approximately 60 hours to sign 15 inputs.
+
+This attack slows down the sweeping schedule but given that the active wallet is
+not performing redemptions or moving funds, it does not affect the funds already
+bridged.
+
+In the first release, the client should assume no more than 3% of the network is
+corrupted and retry the signing of a single input no more than 10 times. If the
+signing of an input failed 10 times, the entire sweep is stopped. This puts the
+group in danger of a fraud challenge from one of the group members but the group
```

Comment:
what's the fraud challenge here? I thought that operators didn't have to sweep deposits